File: labs/poo/lab04/Java/src/Gerente.java
public class Gerente extends Funcionario implements GerenteInterface {
    private int numero_de_equipes;

    public int getNumero_de_equipes() {
        return numero_de_equipes;
    }

    public void setNumero_de_equipes(int numero_de_equipes) {
        this.numero_de_equipes = numero_de_equipes;
    }

    public Gerente(int numero_de_equipes, String nome, int salario_base) {
        super(nome, salario_base);
        this.numero_de_equipes = numero_de_equipes;
    }

    @Override
    public int bonus() {
        return this.numero_de_equipes * 500;
    }

    @Override
    public String descricao() {
        return "Gerente: " + this.getNome() + "\nSalario: " + this.getSalario_base() + "\nNumero de equipes: "
                + this.numero_de_equipes + "\nBonus: " + this.bonus();
    }

}

File: labs/poo/lab04/cpp/Gerente.cpp
#include "Gerente.hpp"

Gerente::Gerente(int numero_equipes, string nome, int salario_base) : numero_equipes(numero_equipes), Funcionairo(nome, salario_base) {};

int Gerente::bonus()
{
    return numero_equipes * 500;
}

int Gerente::getNumeroEquipes() const
{
    return numero_equipes;
}

void Gerente::setNumeroEquipes(int numero_equipes)
{
    this->numero_equipes = numero_equipes;
}

string Gerente::Descricao()
{
    return "Nome: " + getNome() + "\nSalario: " + to_string(getSalarioBase()) + "\nNumero de equipes: " + to_string(getNumeroEquipes()) + "\nBonus: " + to_string(bonus());
}